### PR TITLE
CAMEL-11597 Support for ElasticSearch Client Authentication 

### PR DESCRIPTION
**Pull request by fharms**
Adding properties for setting the username and password for client authentication to the ElasticSearch cluster.

- Properties for adjust timeouts etc. Useful when connecting to the ElasticSearch cloud. 
- Allow to override the client from the component for advanced use cases, eg like using truststore 
- Upgrade ELK to 5.5

Unfortunately I have not found a good way to unittest with XPACK client because it require it can be found on the classpath. Adding XPACK client as a test artifact require we add a new remote repository, so that could be a way to make it testable. 
```xml
<repository>
    <id>elasticsearch-releases</id>     
    <url>https://artifacts.elastic.co/maven</url>
    <releases>
      <enabled>true</enabled>
    </releases>
    <snapshots>
      <enabled>false</enabled>
    </snapshots>
</repository>
```
However I did run manual tests, but I'm open for suggestions. But before using a lot of time you may consider replace the client with the new REST High level API in the near future.

https://discuss.elastic.co/t/can-use-xpack-transporter-client-to-access-a-basic-elasticsearch-server/94815

### CODE REVIEW

File: components/camel-elasticsearch5/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchProducer.java [hunk 1]
```diff
@@ -208,37 +210,53 @@ public void process(Exchange exchange) throws Exception {
...
-    private Settings getSettings() {
-        return Settings.builder()
-                .put("cluster.name", configuration.getClusterName())
-                .put("client.transport.ignore_cluster_name", false)
-                .put("client.transport.sniff", configuration.getClientTransportSniff())
-                .build();
+    private TransportClient createClient() {
+        final Settings.Builder settings = getSettings();
+        try {
+            Class clazz = Class.forName("org.elasticsearch.xpack.client.PreBuiltXPackTransportClient");
```
Comment (davsclaus) [inline components/camel-elasticsearch5/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchProducer.java]: Use Camel's ClassResolver API, eg from CamelContext you can get that to load classes from classpath.

File: components/camel-elasticsearch5/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchProducer.java [hunk 2]
```diff
@@ -208,37 +210,53 @@ public void process(Exchange exchange) throws Exception {
...
+            Constructor<?> ctor = clazz.getConstructor(Settings.class, Class[].class);
+            settings.put("xpack.security.user", configuration.getUser() + ":" + configuration.getPassword())
+                .put("xpack.security.transport.ssl.enabled", configuration.getEnableSSL());
+            return (TransportClient) ctor.newInstance(new Object[] {settings.build(), new Class[0]});
+        } catch (Exception e) {
+            LOG.info("XPack Client was not found on the classpath, using the standard client");
```
Comment (davsclaus) [inline components/camel-elasticsearch5/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchProducer.java]: Should maybe be in DEBUG mode as most people will not use XPack Client. Instead you revert this and do an INFO logging if you can create the XPack Client

File: components/camel-elasticsearch5/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchProducer.java (hunk 1 above)
Comment (fharms) [inline components/camel-elasticsearch5/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchProducer.java]: How do I get the CamelContext, from the endpoint passed into the produce?

Comment (davsclaus) [inline components/camel-elasticsearch5/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchProducer.java]: They all have a getCamelContext og getEndpoint where you can get it from.

**Pull request closed without merging**